src/Bridge/OpenAI/DallE/ContractExtension.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Bridge\OpenAI\DallE;
+
+use PhpLlm\LlmChain\Bridge\OpenAI\DallE;
+use PhpLlm\LlmChain\Model\Model;
+use PhpLlm\LlmChain\Platform\Contract\Extension;
+
+final class ContractExtension implements Extension
+{
+    public function supports(Model $model): bool
+    {
+        return $model instanceof DallE;
+    }
+
+    public function registerTypes(): array
+    {
+        return [
+            'string' => 'handleInput',
+        ];
+    }
```

Comment:
couldn't we do this using an attribute like the event listeners do?

Reply:
reverted back to plain Normalizer since i had to start doing recursive object traversing and array handling myself. not worth it at the moment.